src/components/ModalWithForm.js

```diff
@@ -17,6 +17,15 @@ export default class ModalWithForm extends Modal {
     return inputValues;
   }
 
+  setAvatarUrl(avatarUrl) {
+    console.log(this._modalForm);
+    const avatarInput = this._modalForm.querySelector("input[name='avatar']");
+    console.log(avatarInput);
+    if (avatarInput) {
+      avatarInput.value = avatarUrl;
```

Comment:
* Please, delete the hardcoded `setAvatarUrl` from **universal** `ModalWithForm`


You need to set everything in `UserInfo.js` (and you do it already there)


And you don't need to insert the link into the input. There is not need to do that because you can't edit the avatar link. You can only replace it